src/lang/util.ts

```diff
@@ -69,3 +71,9 @@ export function isLiteral(e: any): e is Literal {
 export function isBinaryExpression(e: any): e is BinaryExpression {
   return e && e.type === 'BinaryExpression'
 }
+
+export function isLiteralValueNotStringAndBoolean(
+  e: any
```

Comment:
The input can still be typed as a literal though right?

Reply:
pushed a fix.